build.xml

```diff
@@ -0,0 +1,163 @@
+
+    <property name="checkstyle.version" value="7.8.2" />
+
+    <property environment="env"/>
+
+    <property name="mx.cmd" value="${basedir}/mx/mx" />
+
+    <property name="build.dir"   value="build"/>
+    <property name="classes.dir" value="${build.dir}/classes"/>
+
+    <path id="project.classpath">
+        <pathelement location="${classes.dir}" />
+        <pathelement location="${test.dir}" />
+        <pathelement location="${sdk.build}/graal-sdk.jar" />
+        <pathelement location="${sdk.build}/word-api.jar" />
+        <pathelement location="${lib.dir}/junit-${junit.version}.jar" />
+        <pathelement location="${lib.dir}/hamcrest-core-1.3.jar" />
+        <pathelement location="${truffle.build}/truffle-api.jar" />
+        <pathelement location="${truffle.build}/truffle-dsl-processor.jar" />
+    </path>
+
+    <target name="clean" description="Remove build directories and generated code">
+        <delete dir="${build.dir}"/>
+        <delete dir="${src_gen.dir}"/>
+    </target>
+    
+    <target name="clobber" description="Do clean, and also clean truffle build" depends="clean">
+        <exec executable="${mx.cmd}" dir="${truffle.dir}">
+            <arg value="clean"/>
+        </exec>
+        <exec executable="${mx.cmd}" dir="${sdk.dir}">
+            <arg value="clean"/>
+        </exec>
+    </target>
+
+    <target name="eclipseformat">
+      <pathconvert pathsep=" " property="javafiles">
+        <fileset dir="${src.dir}">
+          <include name="**/*.java"/>
+        </fileset>
+        <fileset dir="${test.dir}">
+          <include name="**/*.java"/>
+        </fileset>
+      </pathconvert>
+      <exec executable="${env.ECLIPSE_EXE}" dir="${basedir}">
+          <arg value="-nosplash"/>
+          <arg value="-application"/>
+          <arg value="-consolelog"/>
+          <arg value="-data"/>
+          <arg value="${basedir}"/>
+          <arg value="-vm"/>
+          <arg value="${env.JAVA_HOME}/bin/java"/>
+          <arg value="org.eclipse.jdt.core.JavaCodeFormatter"/>
+          <arg value="-config"/>
+          <arg value="${basedir}/.settings/org.eclipse.jdt.core.prefs"/>
+          <arg line="${javafiles}"/>
+      </exec>
+      <exec executable="git" dir="${basedir}" failonerror="true">
+          <arg value="status" />
+          <arg value="*.java" />
+      </exec>
+      <exec executable="git" dir="${basedir}" failonerror="true">
+          <arg value="diff-index" />
+          <arg value="--quiet" />
+          <arg value="HEAD" />
+      </exec>
+    </target>
+    
+    <target name="check-graal-available">
+        <available file="graal/.git" property="graal.present"/>
+    </target>
+    <target name="graal-repo" depends="check-graal-available"
+            unless="graal.present">
+      <exec executable="${basedir}/.graal-git-repo">
+        <arg value="checkout" />
+      </exec>
+    </target>
+
+    <target name="graal" depends="graal-repo">
```

Comment:
At the moment, it is only compiling Truffle, yes.
But we might need more, for instance the Graal SDK stuff, don't know yet.
So, I just named the target after the repo, which is called 'graal'.